### DISCUSSION TITLE
##! mode: xxxx, why is ##! used?

### DISCUSSION THREAD

**Discussion by Pucksnok** [missionpinball/missionpinball#62] [Q&A]
Hello.
I have tried and search but can't find any answer... when I check the documentation often the modes have a code like this

example:
```
##! mode: choose_pokemon
```

Do ##! have any real purpose? Is it used by MPF in any way? Or is it just like using # ? Then why use ##! ?
If someone can link to it in the documentation it would be great because I really tried finding anything about it and I know I also tried finding information about it when I did the tutorial. Now I already asked so many stupid questions already so I thought it's time to ask this as well. :D 

Cheers.

**Comment by Stefku** [comment]
In my understanding it is used just in documentation to define in which file the source code belongs. Otherwise its hard to differentiate.

**Reply by Pucksnok** [reply]
Ok, thx. Seen it other people's code so thought it may have some deeper meaning.
 Cheers.
